.github/workflows/kogito-apps-downstream-pr.yml

```diff
@@ -0,0 +1,59 @@
+name: Kogito Apps Downstream Build
+
+on: [pull_request]
+
+jobs:
+  apps-build:
+    timeout-minutes: 120
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        java-version: [11]
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
+    name: Maven Build
+    steps:
+      - name: Disk space report before modification
+        shell: bash
+        run: |
+          echo "Available storage:"
+          df -h
+      # Inspired to maximize-build-space action https://github.com/easimon/maximize-build-space
+      - name: Free disk space (remove dotnet, android and haskell)
+        shell: bash
+        run: |
+          sudo rm -rf /usr/share/dotnet
+          sudo rm -rf /usr/local/lib/android
+          sudo rm -rf /opt/ghc
+      - name: Disk space report after modification
+        shell: bash
+        run: |
+          echo "Available storage:"
+          df -h
+      - name: Set up JDK
+        uses: actions/setup-java@v1
+        with:
+          java-version: ${{ matrix.java-version }}
+      # See https://docs.github.com/en/actions/guides/building-and-testing-java-with-maven#caching-dependencies
+      - name: Cache Maven packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+      - name: Build Chain ${{ matrix.java-version }}
+        id: build-chain
+        uses: kiegroup/github-action-build-chain@v2.5.7
+        with:
+          definition-file: https://raw.githubusercontent.com/${GROUP}/kogito-pipelines/${BRANCH}/.ci/pull-request-config.yaml
```

Comment:
I am not an expert at CI, but I have a suspicion that this will not work. While both Kogito and Optaplanner have `master` branches (for now), our release branches differ. Where Kogito uses 1.x, we use 8.x. Reusing `${BRANCH}` therefore can not work, I think. (Unless there is some magic there that I am ignorant of.)

Reply:
You are right, for `master` it works well while for branches it will require some additional work: it is already possible to provide a static remapping (like we [do for 7.x](https://github.com/kiegroup/droolsjbpm-build-bootstrap/blob/d3aa20ac2176b34c0a5f62dbeb00ba75b3f3de28/.ci/project-dependencies.yaml#L36)).
I already asked @Ginxo to think about an improvement for `github-action-build-chain` where the branch mapping policies is more advanced. I expect it will take some days/weeks to have it solved.
For now this integration works only if the PR is targeting `master` (so not for backports)

Reply:
Makes sense, thanks for the explanation.

Reply:
@triceo @danielezonca this is the issue to improve it https://github.com/kiegroup/github-action-build-chain/issues/148
I will work on it as soon as I adapt kogito-tooling nightly.